### PR TITLE
Desert Region: world generation changes

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary
Mods "Desert Region: world generation changes"

#### Purpose of change

Continues desert region revival project.

#### Describe the solution

Changes are focused on improving world generation:

* Re-added rivers, made lakes and rivers completely dry
  * added mudcracks to indicate where water once was
  * ~~the implementation of this was somewhat primitive; right now every river and lake OM terrain uses a 24x24 chunk of mudcrack as its map, which looks... bad.~~
  * replaced hardcoded river banks and lake shores with nested chunks
* Renamed sparse forests to 'bushland', dense forests to 'woodland'
* Re-added forest trails with configured settings
* Ported newer city buildings from vanilla
  * except the textile mills because they're based on buildings from NE
* Improved mesa and canyon mutable generation
  * fixed issue mentioned in #57902, now they don't rely on automatic roofs
  * much smoother and natural-looking walls
  * linear canyons; still not perfect, but much better than the old, circular ones

#### Describe alternatives you've considered

~~I've tried using conditional nested chunks to fix shorelines, but that got way too convoluted thanks to all the variables.
My next idea was to create a map extra - similar to `mx_reed` - that converts water into mudcrack, and apply it to every river bank and lake shore. That requires C++ however.~~

#### Testing
Loaded a world with Desert Region.

#### Additional context

<details>

  <summary>Screenshots:</summary>
Forest trails, shrubland, woodland, dry rivers, improved canyon generation:

kép

Smoother mesa walls:

kép

Lake shore:

kép

</details>

### CODE REVIEW

File: data/mods/desert_region/mapgen/waterbody.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+[
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [
+      "lake_bed",
+      "lake_shore",
+      "river_center",
```
Comment (NetSysFire) [inline data/mods/desert_region/mapgen/waterbody.json:8]: I believe you could make the dried out rivers prettier by adding a nested mapgen of a blotch of clay or recesses, with a low chance of some puddles in the middle, only spawning in the rivers center. Same for lakes perhaps. Then making river or lake shores mostly mudcracks, sand and dirt with maybe a bit more than usual vegetation.

**Comment by John-Candlebury** [comment]
Take note that Ravines and Rivers/lakes dont interact nicely with each other (since ravines are designed principally for the Aftershock Exoplanet which has no liquid water). 

You'd either need to expand the generation code so they merge properly, or disable one or the other.

File: data/mods/desert_region/mapgen/waterbody.json (hunk 1 above)
Comment (LovamKicsiGazsii) [reply data/mods/desert_region/mapgen/waterbody.json:8]: I think the simplistic river centers look *okay* for now, I am more concerned about the overly rectangular shorelines. Turning the shore into a mixture of sand, dirt, and mudcrack would not fix the harsh seams between either the center and the shore or the shore and the land, which is my main issue.

Comment (LovamKicsiGazsii) [reply data/mods/desert_region/mapgen/waterbody.json:8]: It's a problem for another day then.  Thank you for your suggestion, I'll consider it.

Comment (LovamKicsiGazsii) [reply data/mods/desert_region/mapgen/waterbody.json:8]: Nevermind, just realized else_chunks exist.  I'll make a commit here once I'm done.

**Pull request merged**